src/algorithms/models/edgecost.ts

```diff
@@ -7,18 +9,111 @@ class EdgeCost {
    * @param ecs EdgeCost[] costs to be combined
    * @return EdgeCost Combined cost of all ecs
    */
+
   static combine(...ecs: EdgeCost[]): EdgeCost {
     const res = new EdgeCost();
     for (const ec of ecs) {
       res.distance += ec.distance;
+      res.time += ec.time;
+      res.road_cost += ec.distance * ec.road_cost;
     }
+
+    res.road_cost /= res.distance;
```

Comment:
Cool, my original idea was to have something like sum variables but this is actually better!

Reply:
![the-most-interesting-man-in-the-world](https://user-images.githubusercontent.com/2762143/48968798-10a3db00-eff5-11e8-9242-40243c2f78d1.jpg)